### PR TITLE
React DP-11634: MainNav a11y

### PR DESCRIPTION
**Pull request by avrilpearl**
## Description
Update the MainNav molecule for better a11y, enabling users to tab through the navigation menu. Also updated the svg-sprite-loader package to permit Icon component

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-11634)

## Steps to Test
1. `cd react`
2. `npm install`
3. `npm start`
4. go to http://localhost:6006/iframe.html?selectedKind=organisms&selectedStory=Header and confirm that you can tab through and on enter expand main nav like video below.

## Screenshots
main-nav

### CODE REVIEW

File: react/src/components/molecules/MainNav/index.js [hunk 1]
```diff
@@ -104,7 +105,7 @@ class MainNav extends Component {
               id: liId
             };
             return(
-              <li {...liClasses} onMouseEnter={(e) => this.mouseOver(e)} onMouseLeave={(e) => this.mouseOut(e)}>
+              <li {...liClasses} role="presentation" onMouseEnter={(e) => this.mouseOver(e)} onMouseLeave={(e) => this.mouseOut(e)}>
```
Comment (avrilpearl) [inline react/src/components/molecules/MainNav/index.js]: bind function to this for opening sub nav on enter and closing on escape...
```suggestion
               <li {...liClasses} role="presentation" onMouseEnter={(e) => this.mouseOver(e)} onMouseLeave={(e) => this.mouseOut(e)} onKeyDown={(e) => this.onKeyDown(e)}>
```

File: react/src/components/molecules/MainNav/index.js [hunk 2]
```diff
@@ -30,10 +30,9 @@ class MainNav extends Component {
   render() {
```
Comment (avrilpearl) [inline react/src/components/molecules/MainNav/index.js:41]: add  handler for keys pressing above this...
```
  onKeyDown(e) {
    console.log(e.keyCode)
    if(e.keyCode === 13){
      this.setState({
        navSelected: e.currentTarget.id
      });
    }
    if(e.keyCode === 27){
      this.setState({
        navSelected: -1
      });
    }
  }
```

File: react/src/components/molecules/MainNav/index.js [hunk 3]
```diff
@@ -30,10 +30,9 @@ class MainNav extends Component {
...
     return(
       <section className="ma__main-nav">
-        <ul className="ma__main-nav__items js-main-nav">
+        <ul className="ma__main-nav__items js-main-nav" role="menubar">
           {this.props.mainNav.map((item, index) => {
             const buttonId = `button${index}`;
-            const menuId = `menu${index}`;
             const liId = `li${index}`;
             const topItemClasses = ['ma__main-nav__item'];
             const { navSelected } = this.state;
```
Comment (avrilpearl) [inline react/src/components/molecules/MainNav/index.js:53]: on line 41, create update is-open and is-closed to react specific classname to enable is-closed to be set to display none on desktop. eg
screen shot 2018-12-19 at 5 38 59 pm

**Review by smurrayatwork — APPROVED** [review]

File: react/src/components/molecules/MainNav/index.js [hunk 4]
```diff
@@ -14,97 +15,103 @@ class MainNav extends Component {
...
     return(
       <section className="ma__main-nav">
-        <ul className="ma__main-nav__items js-main-nav">
+        <ul className="ma__main-nav__items js-main-nav" role="menubar">
           {this.props.mainNav.map((item, index) => {
+            const topItemClasses = classNames({
+              'ma__main-nav__item': true,
+              'is-active': item.active,
+              'has-subnav js-main-nav-toggle': item.subNav,
+              'js-main-nav-top-link': !item.subNav
```
Comment (clairesunstudio) [inline react/src/components/molecules/MainNav/index.js]: looks like these .js- classes are not needed in react

**Review by clairesunstudio — APPROVED** [review]
Looks good! matching solutions in the ticket! It's working differently than mass.gov, is this solution better than what's currently in patterlab @ygannett ?

File: react/src/components/molecules/MainNav/index.js (hunk 4 above)
Comment (avrilpearl) [reply react/src/components/molecules/MainNav/index.js]: removed!

**Pull request merged**